### PR TITLE
fix(aria-allowed-attr): flag attributes that are not supported with certain roles

### PR DESCRIPTION
**Pull request by straker**
I initially thought to make this it's own check, but the `aria-unsupported-attrs` check made the most sense in the end.

Closes issue: #2558

### CODE REVIEW

File: lib/checks/aria/aria-unsupported-attr-evaluate.js [hunk 1]
```diff
@@ -1,7 +1,36 @@
...
+import { validateAttr, getRole } from '../../commons/aria';
 import standards from '../../standards';
 import matches from '../../commons/matches';
 
+/**
+ * Check that an element does not use any unsupported ARIA attributes for it's particular role.
+ * @param {VirtualNode} virtualNode
+ * @return {String[]}
+ */
+function getUnsupportedRoleAttrs(virtualNode) {
```
Comment (WilcoFiers) [inline lib/checks/aria/aria-unsupported-attr-evaluate.js:10]: Kinda think this function should be on commons.aria instead.

File: test/integration/rules/aria-allowed-attr/failures.html [hunk 2]
```diff
@@ -20,4 +20,5 @@
 <div role="subscript" aria-labelledby="value" id="fail20">fail</div>
 <div role="superscript" aria-label="value" id="fail21">fail</div>
 <div role="superscript" aria-labelledby="value" id="fail22">fail</div>
+<fieldset aria-required="false" id="fail23">fail</fieldset>
```
Comment (WilcoFiers) [inline test/integration/rules/aria-allowed-attr/failures.html:23]: I think we're a little too hasty with this. This particular example will definitely not cause accessibility issues. aria-required=false does nothing. Even with aria-required=true... if there is some other indication that the element is required, I don't think we can call that out as a failure of WCAG TBH.

I feel like I should have spotted that a little sooner, but I think this is something that needs review at best, and I would argue only if it's aria-required=true, and the form controls in the fieldset don't have required or aria-required.

**Review by WilcoFiers — CHANGES_REQUESTED** [review]

File: lib/standards/aria-roles.js [hunk 3]
```diff
@@ -479,6 +479,7 @@ const ariaRoles = {
       'aria-expanded',
       'aria-orientation'
     ],
+    unsupportedAttrs: ['aria-required'],
```
Comment (WilcoFiers) [inline lib/standards/aria-roles.js:482]: Are you sure this doesn't work on any radiogroup? I only tested radiogroup on a fieldset with VO. I'd like to have more test data before we make this change.

File: lib/commons/aria/get-unsupported-role-attrs.js [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+      ).length;
+
+      if (
+        role === 'radiogroup' &&
+        vNode.attr('aria-required') === 'true' &&
+        !requiredDescendants
+      ) {
+        return true;
+      }
+    }
```
Comment (WilcoFiers) [inline lib/commons/aria/get-unsupported-role-attrs.js:44]: This seems like it should be its own function somewhere, not embedded into this one. Sticking this here interrupts the reading flow. Maybe just put this below his function?

**Review by WilcoFiers — CHANGES_REQUESTED** [review]

File: lib/commons/aria/get-unsupported-role-attrs.js (hunk 4 above)
Comment (straker) [reply lib/commons/aria/get-unsupported-role-attrs.js:44]: This is in the same code style as https://github.com/dequelabs/axe-core/blob/develop/lib/checks/aria/aria-valid-attr-value-evaluate.js#L39. Should we add a tech debt ticket to update that one?

**Comment by straker** [comment]
Closing as this is just a single use case with a lot of special logic behind it

**Pull request closed without merging**